src/IdentityServer4/Services/DefaultRefreshTokenService.cs

```diff
@@ -99,55 +99,71 @@ public DefaultRefreshTokenService(ISystemClock clock, IRefreshTokenStore refresh
 
             if (client.RefreshTokenUsage == TokenUsage.OneTimeOnly)
             {
-                _logger.LogDebug("Token usage is one-time only. Generating new handle");
-
-                // delete old one
-                await RefreshTokenStore.RemoveRefreshTokenAsync(handle);
-
-                // create new one
-                needsCreate = true;
-            }
-
-            if (client.RefreshTokenExpiration == TokenExpiration.Sliding)
-            {
-                _logger.LogDebug("Refresh token expiration is sliding - extending lifetime");
-
-                // if absolute exp > 0, make sure we don't exceed absolute exp
-                // if absolute exp = 0, allow indefinite slide
-                var currentLifetime = refreshToken.CreationTime.GetLifetimeInSeconds(Clock.UtcNow.UtcDateTime);
-                _logger.LogDebug("Current lifetime: " + currentLifetime.ToString());
-
-                var newLifetime = currentLifetime + client.SlidingRefreshTokenLifetime;
-                _logger.LogDebug("New lifetime: " + newLifetime.ToString());
-
-                // zero absolute refresh token lifetime represents unbounded absolute lifetime
-                // if absolute lifetime > 0, cap at absolute lifetime
-                if (client.AbsoluteRefreshTokenLifetime > 0 && newLifetime > client.AbsoluteRefreshTokenLifetime)
-                {
-                    newLifetime = client.AbsoluteRefreshTokenLifetime;
-                    _logger.LogDebug("New lifetime exceeds absolute lifetime, capping it to " + newLifetime.ToString());
-                }
-
-                refreshToken.Lifetime = newLifetime;
-                needsUpdate = true;
-            }
-
-            if (needsCreate)
-            {
+                needsCreate = await RemoveRefreshToken(handle);
                 handle = await RefreshTokenStore.StoreRefreshTokenAsync(refreshToken);
```

Comment:
Should all of these `async`  tasks also have `.ConfigureAwait(false)`? It doesn't appear the codebase has much them, but maybe it's OK to add them as one refactor anyhow. This doesn't only help with "GUI systems", but also with systems like https://github.com/dotnet/orleans (maybe even ASP.NET Core).

Reply:
`ConfigureAwait(false)` is not necessary in ASP.NET Core, as it [doesn't have a synchronization context](https://stackoverflow.com/a/42058544/1746803).